### PR TITLE
Copter: Move "land_run_horizontal_control()" to use position controller for Prec Land 

### PR DESCRIPTION
**Pull request by rishabsingh3003**
This is a rework of #18111
It keeps the Loiter control for repositioning however uses the updated position controller for Prec Landing

### CODE REVIEW

**Comment by amilcarlucas** [comment]
How extensively has this been tested? How safe is it?

**Comment by rishabsingh3003** [comment]
@amilcarlucas the last linked PR was extensively tested by me, and two other users who were interested. 
This one has a few changes from the last PR and is only tested on Simulators
I'll be doing tests this weekend

**Comment by amilcarlucas** [comment]
How did the tests go?

**Comment by amilcarlucas** [comment]
ping

**Comment by rishabsingh3003** [comment]
@amilcarlucas sorry for leaving this one behind. I am going out to test this some more tomorrow. Fixed the conflicts

File: libraries/AC_PrecLand/AC_PrecLand.cpp [hunk 1]
```diff
@@ -162,6 +162,13 @@ const AP_Param::GroupInfo AC_PrecLand::var_info[] = {
     // @Units: m
     AP_GROUPINFO("ALT_MAX", 16, AC_PrecLand, _sensor_max_alt, 8),
 
+    // @Param: MOVING
+    // @DisplayName: Precision Landing Target stationary/moving
+    // @Description: Precision Landing Target stationary/moving
+    // @Values: 0:Stationary, 1:Moving
+    // @User: Advanced
+    AP_GROUPINFO("MOVING", 17, AC_PrecLand, _moving, 0),
```
Comment (rmackay9) [inline libraries/AC_PrecLand/AC_PrecLand.cpp:170]: Let's change this to an OPTIONS parameter so we can shove more features in here in the future.

**Comment by rmackay9** [comment]
After the change to the "MOVING" -> "OPTIONS" parameter naming let's merge this

**Comment by rmackay9** [comment]
Merged, thanks!  @rishabsingh3003 will submit a follow-up PR to rename MOVING->OPTIONS.

**Pull request merged**